tests/components/prometheus/test_init.py

```diff
@@ -137,15 +159,19 @@ async def test_setup_enumeration(
         suggested_object_id="outside_temperature",
         original_name="Outside Temperature",
     )
-    set_state_with_entry(hass, sensor_1, 12.3, {})
+    state = 12.3
+    set_state_with_entry(hass, sensor_1, state, {})
     assert await async_setup_component(hass, prometheus.DOMAIN, {prometheus.DOMAIN: {}})
 
     client = await hass_client()
     body = await generate_latest_metrics(client)
-    assert (
-        'homeassistant_sensor_temperature_celsius{domain="sensor",'
-        'entity="sensor.outside_temperature",'
-        'friendly_name="Outside Temperature"} 12.3' in body
+    MetricsTestHelper._perform_sensor_metric_assert(
+        "homeassistant_sensor_temperature_celsius",
+        "12.3",
+        "Outside Temperature",
+        "outside_temperature",
+        body,
+        device_class=SensorDeviceClass.TEMPERATURE,
```

Comment:
Looking at the diff here, I think for an individual assertion the new version is less legible and clear.

The previous version was very simple: it confirmed that a particular string was present in a list.

In the new version, usage of positional arguments makes it difficult to understand what those arguments are. Perhaps the helpers should accept a `dict` with label keys & values, to make it more obvious what is being checked? E.g.:
```python
assert_metric_present(body, "homeassistant_sensor_temperature_celsius", 12.3, dict(
  entity="sensor.outside_temperature",
  domain="sensor",
  friendly_name="Outside Temperature",
  object_id="outside_temperature",
  device_class="temperature",
))
```